src/auto-merge.ts

```diff
@@ -1,18 +1,72 @@
 import type { Context } from "@actions/github/lib/context";
 
+import {
+  NotDependabotPrError,
+  NotValidSemverError,
+  ParseError,
+  ReviewAlreadyPendingError,
+  UnsupportedFeatureError,
+} from "./errors";
 import * as github from "./lib/github";
+import * as parse from "./lib/parse";
+import * as version from "./lib/version";
 
-export class NotDependabotPrError extends Error {}
+export default async function autoMerge(
+  context: Context,
+  rawDisallowList: string,
+  rawReviewers: string,
+): Promise<["OK" | "NOK", string]> {
```

Comment:
Might be nice to type the return to make what the string is more explicit with something like:

```typescript
type AutoMergeResult = {
  status: "OK" | "NOK",
  errorMessage: string
}
```

Though we go from an array to an object, but still simple to destructure.

Reply:
Yep, good idea. It's been implemented instead of the tuple which is less "Javascripty" I guess 😄